synapse/rest/client/v2_alpha/keys.py

```diff
@@ -51,23 +54,47 @@ class KeyUploadServlet(RestServlet):
       },
     }
     """
-    PATTERNS = client_v2_patterns("/keys/upload/(?P<device_id>[^/]*)", releases=())
+    PATTERNS = client_v2_patterns("/keys/upload(/(?P<device_id>[^/]+))?$",
+                                  releases=(), v2_alpha=False)
```

Comment:
yeah he does. https://github.com/torhve/weechat-matrix-protocol-script/blob/master/matrix.lua

and I see he's using v2_alpha

FUCK IT

this is why we shouldn't publish apis on v2_alpha until they are stable.